python/hail/methods/impex.py

```diff
@@ -764,13 +764,179 @@ def import_gen(path, sample_file=None, tolerance=0.2, min_partitions=None, chrom
+    ``NA#####``. This file has no header line, and the sample ID is hidden in a
+    field with other information.
+
+    .. code-block: text
+
+        $ cat data/samples3.tsv
+        1kg_NA12345   female
+        1kg_NA12346   male
+        1kg_NA12348   female
+        pgc_NA23415   male
+        pgc_NA23418   male
+
+    To import:
+
+    >>> annotations = (hc1.import_table('data/samples3.tsv', no_header=True)
+    ...                   .annotate('sample = f0.split("_")[1]')
+    ...                   .key_by('sample'))
+
+    **Notes**
+
+    The `impute` parameter tells Hail to scan the file an extra time to gather
+    information about possible field types. While this is a bit slower for large
+    files, (the file is parsed twice), the convenience is often worth this cost.
+
+    The `delimiter` parameter is either a delimiter character (if a single
+    character) or a field separator regex (2 or more characters). This regex
+    follows the `Java regex standard
+    <http://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html>`_.
+
+    .. note::
+
+        Use ``delimiter='\\s+'`` to specify whitespace delimited files.
+
+    `comment` optionally causes Hail to skip any line that starts with the given
+    string. Passing ``comment='#'`` will skip any line beginning in a pound
+    sign, for example.
+
+    `missing` defines the representation of missing data in the table.
+
+    .. note::
+
+        The `comment` and `missing` parameters are **NOT** regexes.
+
+    The `no_header` parameter indicates that the file has no header line. If
+    this option is passed, then the column names will be `f0`, `f1`,
+    ... `fN` (0-indexed).
+
+    The `types` parameter allows the user to pass the types of columns in the
+    table. This is a dict keyed by :obj:`str`, with :class:`.Type`
+    values. See the examples above for a standard usage. Additionally, this
+    option can be used to override type imputation. For example, if a column in
+    a file refers to chromosome and does not contain any sex chromosomes, it
+    will be imputed as an integer, while most Hail methods expect chromosome to
+    be passed as a string. Using the ``impute=True`` mode and passing
+    ``types={'Chromosome': TString()}`` will solve this problem.
+
+    The `min_partitions` parameter sets the minimum number of partitions (level
+    of sharding) of an imported table. The default partition size depends on
+    file system and a number of other factors (including the ``min_block_size``
+    of the hail context), but usually is between 32M and 128M.
+
+    Parameters
+    ----------
+
+    paths: :obj:`str` or :obj:`list` of :obj:`str`
+        Files to import.
+
+    key: :obj:`str` or :obj:`list` of :obj:`str`
+        Key column(s).
+
+    min_partitions: :obj:`int` or :obj:`None`
+        Minimum number of partitions.
+
+    no_header: :obj:`bool`
+        File has no header and the N columns are named `f0`, `f1`,
```

Comment:
If true, assume the file has no header and name the N columns `f0`, `f1`, ... (0-indexed)